src/models/branch_predictor.py

```diff
@@ -66,21 +66,26 @@ def can_make_prediction_for_node(self, node):
         # We can't make a prediction for a node that doesn't have children
         return len(node.recursive_children()) > 1
 
-    def predict(self, tree, apex_node, text_to_predict, translated_tokenized_text_to_predict):
+    def predict(self, tree, apex_node, text_to_predict, translated_tokenized_text_to_predict, request_record):
         """
         Makes a prediction for text_to_predict for any taxon beneath the apex_node
         Uses the translated_tokenized_text_to_predict to tell the user which words
         it considered important when making the prediction
         """
         node = apex_node
-        result = []
+        results = []
         while self.can_make_prediction_for_node(node):
             model = dirs.open_pickle_file(self.model_path(node))
             vectorizer = dirs.open_pickle_file(self.vectorizer_path(node))
             transformed_text = vectorizer.transform([text_to_predict])
             words_to_explain_choice = nlp.prediction_explanation(vectorizer, transformed_text, translated_tokenized_text_to_predict)
             probabilities = model.predict_proba(transformed_text)
             highest_probability_content_id = model.classes_[np.argmax(probabilities)]
-            result.append({'taxon_content_id': highest_probability_content_id, 'explanation': words_to_explain_choice})
+            probabilities_for_taxon = {}
+            for i, probability in enumerate(probabilities[0]):
```

Comment:
What is `probabilities[0]` here? From what I've read `predict_proba` isn't ordering these by most>least likely...

Reply:
As discussed